### ISSUE TITLE
assuming every macOS user uses brew for openssl is not OK

### ISSUE THREAD

**Issue by jxy** [gambit/gambit#512]
I guess `configure` wants to be helpful and give sensible default, but practically I just see a lot of warnings from `gsc`,

```
ld: warning: directory not found for option '-L/usr/local/opt/openssl@1.1/lib'
```

which I found out was hardcoded here,

https://github.com/gambit/gambit/blob/c584bbcf6d9c66530dd3f9b981726f22a5973fde/configure#L9833-L9835

The simple thing would be to check if the directory exists before including in `LIBS`.

For me I'm just using,
```
if test "$ENABLE_OPENSSL" = yes; then
  LIBS="$LIBS -lssl -lcrypto"
fi
```
because I pass `CFLAGS=-I/opt/local/include LIBS=-L/opt/local/lib` to configure anyway.

**Comment by lassik** [comment]
This is complicated by the fact that some OpenSSL versions shipped by operating systems are outdated. Another complication is that the `openssl` shell command may belong to a different installation of OpenSSL/LibreSSL than the `libssl` library, even on the same computer. Hence it's not reliable to run the shell command to infer things about the library.

Please see my notes at https://misc.lassi.io/2019/chicken-openssl/chicken-openssl-pkg-config.text

**Comment by lassik** [comment]
For MacOS in particular, last I checked it doesn't ship pkg-config `.pc` files for the system OpenSSL/LibreSSL. Neither do some other operating systems. That's a pity.

Given the lack of `.pc` files, and the possibly outdated versions, it would be easier to rely on Homebrew OpenSSL/LibreSSL on Mac. MacOS versions of open source libraries tend to lag behind the upstream ones enough that programming without Homebrew is a bit difficult in general. Is there a particular reason you'd like to avoid using Homebrew packages?

**Comment by gambiteer** [comment]
On 3/25/20 3:29 AM, Lassi Kortela wrote:
> For MacOS in particular, last I checked it doesn't ship pkg-config |.pc| 
> files for the system OpenSSL/LibreSSL. Neither do some other operating 
> systems. That's a pity.
> 
> Given the lack of |.pc| files, and the possibly outdated versions, it 
> would be easier to rely on Homebrew OpenSSL/LibreSSL on Mac. MacOS 
> versions of open source libraries tend to lag behind the upstream ones 
> enough that programming without Homebrew is a bit difficult in general. 
> Is there a particular reason you'd like to avoid using Homebrew packages?

I use fink's openssl. (I still prefer fink.)

Brad

**Comment by jxy** [comment]
There's no practical way to detect all the possible places people/system might install their libraries anyhow.  Just let user pass in their setup via `CFLAGS` and `LIBS` is good enough.  There's `ax_check_openssl` if you want some automation, but the current way is not as half as good.

**Comment by lassik** [comment]
DoesFink install pkg-config `.pc` to go along with libraries? If it puts those in well-known directories, we may be able to add those directories to Gambit's configure script. Then Fink could easily be used as an alternative to Homebrew.

**Comment by mk12** [comment]
I ran into this today trying to using Gambit (via Gerbil) on an M1 Mac. It's unusable unless I make a symlink as done in [this comment](https://github.com/vyzo/gerbil/issues/423#issuecomment-1015726872), which feels like an unsavory hack to me. IMO it would be better to require everyone to specify the path manually, rather than hardcoding it so that it works out of the box for older Intel Macs using Homebrew but not for Apple Silicon Macs.

**Comment by feeley** [comment]
Indeed.  Can you send in a PR?  I don't have an M1 Mac handy to do testing.

**Comment by lassik** [comment]
There are now a few "cloud" providers where one can rent M1 Macs for a low hourly rate. Though at least one of them has a minimum rent time of one day.
